File: cpp/LeetCode3442.cpp
class Solution {
public:
    int maxDifference(string s) {
        unordered_map<char, int> Ds;
        for (char c : s) {
            Ds[c]++;
        }
        
        int min_ = 100;
        int max_ = 0;
        for (auto i : Ds) {
            if (i.second % 2 == 1 && i.second > max_) {
                max_ = i.second;
            } 
            if (i.second % 2 == 0 && i.second < min_) {
                min_ = i.second;
            }
        }
        
        return max_ - min_;
    }
};

File: rust/LeetCode3442.rs
use std::collections::HashMap;

impl Solution {
    pub fn max_difference(s: String) -> i32 {
        let mut ds = HashMap::new();
        
        // Count frequency of each character
        for c in s.chars() {
            *ds.entry(c).or_insert(0) += 1;
        }
        
        let mut min_val = 100;
        let mut max_val = 0;
        
        // Find max odd frequency and min even frequency
        for (_, count) in ds {
            if count % 2 == 1 && count > max_val {
                max_val = count;
            }
            if count % 2 == 0 && count < min_val {
                min_val = count;
            }
        }
        
        max_val - min_val
    }
}